MIGRATION.md

```diff
@@ -231,3 +232,53 @@ updateFavoriteNumber(n: uint64) {
         { ...this.listings.get(Txn.sender),  number: n }
     );
 ```
+
+### ARC4 Types in State
+
+#### TEALScript
+
+In TEALScript, the same types used within method logic can be stored in state or used as state map keys.
+
+```ts
+export type ListingKey = {
+    owner: Address,
+    asset: Asset,
+    nonce: uint64,
+}
+```
+
+```ts
+updateListing(xfer: AssetTransferTxn, nonce: uint64)
+    const key: ListingKey = {
+      owner: this.txn.sender,
+      asset: xfer.xferAsset,
+      nonce: nonce,
+    })
+
+    const listing = this.listings(key).value
+```
+
+#### PuyaTS
+
+In PuyaTS, state can only hold ARC4 types.
```

Comment:
I'm not 100% sure on where PuyaTS support is at, but the puya backed certainly supports many more types in state than just ARC-4 types.

Reply:
Issue is here https://github.com/algorandfoundation/puya-ts/issues/129